apis/apiextensions/v1/xrd_webhooks.go

```diff
@@ -53,27 +46,7 @@ func (in *CompositeResourceDefinition) ValidateCreate() (admission.Warnings, err
 }
 
 // ValidateUpdate is run for update actions.
-func (in *CompositeResourceDefinition) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
-	oldObj, ok := old.(*CompositeResourceDefinition)
-	if !ok {
-		return nil, errors.New(errUnexpectedType)
-	}
-	switch {
-	case in.Spec.Group != oldObj.Spec.Group:
-		return nil, errors.New(errGroupImmutable)
-	case in.Spec.Names.Plural != oldObj.Spec.Names.Plural:
-		return nil, errors.New(errPluralImmutable)
-	case in.Spec.Names.Kind != oldObj.Spec.Names.Kind:
-		return nil, errors.New(errKindImmutable)
-	}
-	if in.Spec.ClaimNames != nil && oldObj.Spec.ClaimNames != nil {
-		switch {
-		case in.Spec.ClaimNames.Plural != oldObj.Spec.ClaimNames.Plural:
-			return nil, errors.New(errClaimPluralImmutable)
-		case in.Spec.ClaimNames.Kind != oldObj.Spec.ClaimNames.Kind:
-			return nil, errors.New(errClaimKindImmutable)
-		}
-	}
+func (in *CompositeResourceDefinition) ValidateUpdate(_ runtime.Object) (admission.Warnings, error) {
```

Comment:
Should we keep this for K8s versions not supporting CEL yet? For now at least.

Reply:
I would rather find a more organic approach, we could think of loading the crds at runtime and use them to validate resources, so that we still have a way to validate (and test them) from code.

Reply:
I'll work on that as part of this PR, makes sense 👍

Reply:
if this method is empty, do we need validation webhook at all now?

Reply:
> Should we keep this for K8s versions not supporting CEL yet? For now at least.

@lsviben CEL became beta in 1.25, and we do not support 1.24 or lower: https://kubernetes.io/releases/patch-releases/#1-24

Reply:
on this topic, @phisco have you thought through how moving to CEL rules would affect our ability to integrate this validation logic into a development time experience, e.g. a language server? Does it get much harder to have this logic in a language server when it's no longer go library code that can be reused there? I don't know much about the topic myself, but I'm definitely interested in it 😇